lib/pyld/jsonld.py

```diff
@@ -1979,6 +1982,10 @@ def _expand(
                 not (
                     _is_absolute_iri(expanded_property) or
                     _is_keyword(expanded_property))):
+                if options['strict']:
+                    raise ValueError("Unrecognized key %s" % key)
+                elif droppedKeys is not None:
+                    droppedKeys.add(key)
```

Comment:
Returning a value from a function by modifying an argument is quite an unpopular method with Python programmers. This is a side effect; I would discourage it.

Reply:
@anatoly-scherbakov could you refactor this in a way that still fixes the issue?

Reply:
@anatoly-scherbakov I disagree that it is inherently a bad thing for a function to mutate one of its arguments, as long as the contract is clear about it. 
That being said, I agree that having this "output argument" included in the "options" argument (which is clearly an input argument otherwise), is not great.

I don't consider that changing the return value of `expand`, in order to include the information about dropped keys, is desirable (it would break existing code).

I believe that a better solution to be to add a 4th parameter to expand: `dropped=None`. If that parameter is explicitly specified, it must be an (empty?) list, and the contract would be that this list would be populated with the dropped keys.

Reply:
Even if the contract is explicitly specified, this is not what a Python developer is _expecting_ from a library, thus violating the [Principle of Least Astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

I have proposed a PR: https://github.com/digitalbazaar/pyld/pull/186 — but its not ready for merge as it should be really targeted at another PR of mine.